src/Views/IMEView.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <global_include.h>
+
+#include <Views/View.h>
+#include <sys/socket.h>
+
+
+typedef enum IMEViewStatus {
+	IMEVIEW_STATUS_NONE     = SCE_COMMON_DIALOG_STATUS_NONE,
+	IMEVIEW_STATUS_RUNNING  = SCE_COMMON_DIALOG_STATUS_RUNNING,
+	IMEVIEW_STATUS_FINISHED = SCE_COMMON_DIALOG_STATUS_FINISHED,
+	IMEVIEW_STATUS_CANCELED
+}IMEViewStatus;
+
+
+struct IMEViewResult {
+	IMEViewStatus status = IMEVIEW_STATUS_NONE;
+	std::string userText;
+};
+
+
+class IMEView : public View {
+public:
+	IMEView(IMEViewResult *result, const char title[] , const char showText[]);
+	IMEView(IMEViewResult *result, const char title[] , const char showText[],
+		SceUInt32 maxInputLength);
+	IMEView(IMEViewResult *result, const char title[] , const char showText[],
+		const char initialText[]);
+	IMEView(IMEViewResult *result, const char title[] , const char showText[],
```

Comment:
I would suggest having only one constructor with default parameters.

Reply:
Sorry first real project in c++. I'd don't have that much experience. Didn't know default params existed.

Reply:
Welcome to C++ 😃